File: .ai/scripts/GenDictionaryLogos.java
import javax.imageio.ImageIO;
import java.awt.*;
import java.awt.geom.*;
import java.awt.image.BufferedImage;
import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

/**
 * Генератор тематических логотипов Dictionary Edit-форм HRM HuntTech.
 *
 * Рисует 200x200 PNG в палитре HuntTech (фон #172638, символ #f8fafc,
 * акцент #e74c3c) примитивами Java2D и раскладывает во все 7 тем:
 * modules/web/themes/<theme>/icons/dictionaries/<name>.png
 *
 * Запуск (Java 11):
 *   javac GenDictionaryLogos.java && java GenDictionaryLogos
 */
public class GenDictionaryLogos {

    static final int SIZE = 200;
    static final int C = SIZE / 2;

    static final Color BG = new Color(23, 38, 56, 255);
    static final Color FG = new Color(248, 250, 252, 255);
    static final Color ACC = new Color(231, 76, 60, 255);
    static final Color MUT = new Color(148, 163, 184, 255);

    static final String[] THEMES = {
            "halo", "havana", "helium", "hover",
            "hunttech-modern", "hunttech-modern-dark", "hunttech-modern-light"
    };

    interface Painter {
        void paint(Graphics2D g);
    }

    static BufferedImage canvas() {
        BufferedImage img = new BufferedImage(SIZE, SIZE, BufferedImage.TYPE_INT_ARGB);
        Graphics2D g = img.createGraphics();
        g.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
        g.setRenderingHint(RenderingHints.KEY_STROKE_CONTROL, RenderingHints.VALUE_STROKE_PURE);
        g.setColor(BG);
        g.fillRect(0, 0, SIZE, SIZE);
        return img;
    }

    static void ring(Graphics2D g, double cx, double cy, double r, Color c, double w) {
        g.setColor(c);
        g.setStroke(new BasicStroke((float) w, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
        g.draw(new Ellipse2D.Double(cx - r, cy - r, 2 * r, 2 * r));
    }

    static void line(Graphics2D g, double x1, double y1, double x2, double y2, Color c, double w) {
        g.setColor(c);
        g.setStroke(new BasicStroke((float) w, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
        g.draw(new Line2D.Double(x1, y1, x2, y2));
    }

    static void rect(Graphics2D g, double x, double y, double w, double h, Color c) {
        g.setColor(c);
        g.fill(new Rectangle2D.Double(x, y, w, h));
    }

    static void rounded(Graphics2D g, double x, double y, double w, double h, double r, Color c, boolean fill) {
        g.setColor(c);
        RoundRectangle2D rr = new RoundRectangle2D.Double(x, y, w, h, r, r);
        if (fill) {
            g.fill(rr);
        } else {
            g.setStroke(new BasicStroke(8f, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
            g.draw(rr);
        }
    }

    static void ellipse(Graphics2D g, double x, double y, double w, double h, Color c) {
        g.setColor(c);
        g.fill(new Ellipse2D.Double(x, y, w, h));
    }

    static void triangle(Graphics2D g, double x1, double y1, double x2, double y2, double x3, double y3, Color c) {
        g.setColor(c);
        Path2D p = new Path2D.Double();
        p.moveTo(x1, y1);
        p.lineTo(x2, y2);
        p.lineTo(x3, y3);
        p.closePath();
        g.fill(p);
    }

    // ---- iteraction: две круговые стрелки цикла ----
    static void iteraction(Graphics2D g) {
        ring(g, C, C, 62, FG, 10);
        for (double ang : new double[]{Math.toRadians(45), Math.toRadians(225)}) {
            double x1 = C + 62 * Math.cos(ang);
            double y1 = C + 62 * Math.sin(ang);
            double bx1 = C + 52 * Math.cos(ang + 0.18);
            double by1 = C + 52 * Math.sin(ang + 0.18);
            double bx2 = C + 52 * Math.cos(ang - 0.18);
            double by2 = C + 52 * Math.sin(ang - 0.18);
            triangle(g, x1, y1, bx1, by1, bx2, by2, FG);
        }
        ellipse(g, C - 16, C - 16, 32, 32, ACC);
    }

    // ---- specialisation: мишень ----
    static void specialisation(Graphics2D g) {
        ring(g, C, C, 70, FG, 8);
        ring(g, C, C, 46, FG, 8);
        ring(g, C, C, 22, FG, 8);
        ellipse(g, C - 8, C - 8, 16, 16, ACC);
    }

    // ---- country: глобус с меридианами и пин ----
    static void country(Graphics2D g) {
        ring(g, C, C, 62, FG, 8);
        line(g, C, C - 62, C, C + 62, FG, 6);
        line(g, C - 62, C, C + 62, C, FG, 6);
        g.setColor(MUT);
        g.setStroke(new BasicStroke(5f, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
        g.draw(new Arc2D.Double(C - 62, C - 34, 124, 68, 0, 180, Arc2D.OPEN));
        g.draw(new Arc2D.Double(C - 62, C - 34, 124, 68, 180, 180, Arc2D.OPEN));
        g.draw(new Arc2D.Double(C - 34, C - 62, 68, 124, 90, 180, Arc2D.OPEN));
        g.draw(new Arc2D.Double(C - 34, C - 62, 68, 124, -90, 180, Arc2D.OPEN));
        triangle(g, C - 14, C - 88, C + 14, C - 88, C, C - 58, ACC);
        ellipse(g, C - 9, C - 97, 18, 18, ACC);
    }

    // ---- city: силуэты зданий ----
    static void city(Graphics2D g) {
        rect(g, C - 84, C - 46, 54, 108, FG);
        rect(g, C - 74, C - 78, 34, 32, FG);
        for (int dx : new int[]{-68, -46}) {
            for (int dy : new int[]{-30, 0, 30}) {
                rect(g, dx, C + dy - 12, 10, 24, BG);
            }
        }
        rect(g, C - 24, C - 70, 50, 132, MUT);
        for (int dx : new int[]{-14, 6}) {
            for (int dy : new int[]{-52, -24, 4, 32}) {
                rect(g, dx, C + dy - 10, 8, 20, BG);
            }
        }
        rect(g, C + 34, C - 28, 52, 90, FG);
        for (int dx : new int[]{44, 66}) {
            for (int dy : new int[]{-14, 12}) {
                rect(g, dx, C + dy - 10, 10, 20, BG);
            }
        }
    }

    // ---- region: карта с дорогами и пин ----
    static void region(Graphics2D g) {
        g.setColor(FG);
        g.setStroke(new BasicStroke(8f, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
        Path2D map = new Path2D.Double();
        map.moveTo(C - 78, C - 62);
        map.lineTo(C - 12, C - 74);
        map.lineTo(C + 66, C - 46);
        map.lineTo(C + 80, C + 30);
        map.lineTo(C + 20, C + 66);
        map.lineTo(C - 60, C + 52);
        map.lineTo(C - 82, C + 10);
        map.closePath();
        g.draw(map);
        line(g, C - 40, C - 50, C + 30, C + 40, MUT, 7);
        line(g, C - 60, C + 20, C + 50, C - 20, MUT, 7);
        ellipse(g, C - 10, C - 14, 20, 20, ACC);
    }

    // ---- ownershup: документ с ключом ----
    static void ownershup(Graphics2D g) {
        rounded(g, C - 58, C - 78, 116, 148, 10, FG, false);
        line(g, C - 34, C - 40, C + 34, C - 40, FG, 7);
        line(g, C - 34, C - 10, C + 34, C - 10, FG, 7);
        line(g, C - 34, C + 20, C + 10, C + 20, FG, 7);
        ring(g, C - 10, C + 46, 16, ACC, 8);
        line(g, C + 16, C + 40, C + 44, C + 12, ACC, 8);
        line(g, C + 44, C + 12, C + 56, C + 12, ACC, 8);
        line(g, C + 40, C + 16, C + 52, C + 28, ACC, 8);
    }

    // ---- position: бейдж с человечком ----
    static void position(Graphics2D g) {
        rounded(g, C - 66, C - 80, 132, 150, 16, FG, false);
        ellipse(g, C - 20, C - 56, 40, 40, FG);
        g.setColor(FG);
        g.fill(new Arc2D.Double(C - 44, C + 8, 88, 88, 180, 180, Arc2D.PIE));
        ellipse(g, C - 16, C + 44, 32, 16, ACC);
    }

    // ---- file-type: файл с загнутым уголком ----
    static void fileType(Graphics2D g) {
        rounded(g, C - 62, C - 78, 124, 148, 10, FG, false);
        g.setColor(BG);
        Path2D corner = new Path2D.Double();
        corner.moveTo(C + 22, C - 78);
        corner.lineTo(C + 62, C - 78);
        corner.lineTo(C + 62, C - 38);
        corner.lineTo(C + 22, C - 38);
        corner.closePath();
        g.fill(corner);
        line(g, C + 22, C - 78, C + 62, C - 38, FG, 8);
        line(g, C + 22, C - 78, C + 22, C - 38, FG, 8);
        for (int dy : new int[]{-16, 8, 32}) {
            line(g, C - 34, C + dy, C + 30, C + dy, MUT, 7);
        }
        line(g, C - 34, C + 48, C + 10, C + 48, ACC, 7);
    }

    // ---- grade: лестница уровней ----
    static void grade(Graphics2D g) {
        for (int i = 0; i < 4; i++) {
            int y = C + 66 - i * 30;
            int x0 = C - 78 + i * 26;
            Color c = i % 2 == 0 ? FG : MUT;
            line(g, x0, y, C + 70, y, c, 9);
            line(g, x0, y, x0, y - 30, c, 9);
        }
        line(g, C + 70, C - 24, C + 70, C + 66, FG, 9);
        ellipse(g, C + 60, C - 40, 20, 20, ACC);
    }

    // ---- currency: монета с символом ----
    static void currency(Graphics2D g) {
        ring(g, C, C, 70, FG, 10);
        line(g, C + 6, C - 46, C + 6, C + 46, FG, 10);
        line(g, C - 14, C - 24, C + 26, C - 24, FG, 9);
        line(g, C - 14, C + 24, C + 26, C + 24, FG, 9);
        line(g, C + 6, C - 58, C + 6, C - 40, ACC, 9);
        line(g, C + 6, C + 40, C + 6, C + 58, ACC, 9);
    }

    // ---- outstaffing-rates: шкала со стрелкой ----
    static void outstaffingRates(Graphics2D g) {
        rounded(g, C - 76, C - 10, 152, 20, 10, FG, false);
        rounded(g, C - 76, C - 10, 82, 20, 10, FG, true);
        triangle(g, C - 6, C - 52, C + 26, C - 52, C + 10, C - 84, FG);
        line(g, C - 6, C - 52, C + 26, C - 52, FG, 8);
        line(g, C + 10, C - 84, C + 10, C - 52, FG, 8);
        line(g, C - 6, C - 52, C + 10, C - 84, FG, 8);
        line(g, C + 10, C - 84, C + 26, C - 52, FG, 8);
        ellipse(g, C + 2, C - 22, 20, 20, ACC);
    }

    // ---- employee-work-status: часы ----
    static void employeeWorkStatus(Graphics2D g) {
        ring(g, C, C, 70, FG, 10);
        line(g, C, C - 46, C, C, FG, 9);
        line(g, C, C, C + 34, C + 16, FG, 9);
        ellipse(g, C - 9, C - 9, 18, 18, FG);
        ellipse(g, C - 14, C + 46, 28, 28, ACC);
    }

    // ---- sign-icons: звезда ----
    static void signIcons(Graphics2D g) {
        Path2D star = new Path2D.Double();
        for (int i = 0; i < 10; i++) {
            double r = i % 2 == 0 ? 74 : 30;
            double a = Math.PI / 2 + i * Math.PI / 5;
            double x = C + r * Math.cos(a);
            double y = C - 4 + r * Math.sin(a);
            if (i == 0) {
                star.moveTo(x, y);
            } else {
                star.lineTo(x, y);
            }
        }
        star.closePath();
        g.setColor(FG);
        g.setStroke(new BasicStroke(8f, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
        g.draw(star);
        ellipse(g, C - 12, C + 44, 24, 24, ACC);
    }

    public static void main(String[] args) throws Exception {
        java.util.Map<String, Painter> gens = new java.util.LinkedHashMap<>();
        gens.put("iteraction", GenDictionaryLogos::iteraction);
        gens.put("specialisation", GenDictionaryLogos::specialisation);
        gens.put("country", GenDictionaryLogos::country);
        gens.put("city", GenDictionaryLogos::city);
        gens.put("region", GenDictionaryLogos::region);
        gens.put("ownershup", GenDictionaryLogos::ownershup);
        gens.put("position", GenDictionaryLogos::position);
        gens.put("file-type", GenDictionaryLogos::fileType);
        gens.put("grade", GenDictionaryLogos::grade);
        gens.put("currency", GenDictionaryLogos::currency);
        gens.put("outstaffing-rates", GenDictionaryLogos::outstaffingRates);
        gens.put("employee-work-status", GenDictionaryLogos::employeeWorkStatus);
        gens.put("sign-icons", GenDictionaryLogos::signIcons);

        Path base = Paths.get("modules", "web", "themes");
        for (java.util.Map.Entry<String, Painter> e : gens.entrySet()) {
            BufferedImage img = canvas();
            Graphics2D g = img.createGraphics();
            e.getValue().paint(g);
            g.dispose();
            for (String theme : THEMES) {
                Path dir = base.resolve(theme).resolve("icons").resolve("dictionaries");
                Files.createDirectories(dir);
                ImageIO.write(img, "png", dir.resolve(e.getKey() + ".png").toFile());
            }
            System.out.println("OK " + e.getKey() + " " + img.getWidth() + "x" + img.getHeight());
        }
    }
}

File: .ai/scripts/gen-dictionary-logos.py
#!/usr/bin/env python3
"""Генерация тематических логотипов Dictionary Edit-форм HRM HuntTech.

Рисует 200x200 PNG в палитре HuntTech (фон #172638, символ #f8fafc,
акцент #e74c3c) простыми геометрическими примитивами PIL и раскладывает
во все 7 тем: icons/dictionaries/<name>.png.
"""
import math
import os
from PIL import Image, ImageDraw

BG = (23, 38, 56, 255)       # #172638
FG = (248, 250, 252, 255)    # #f8fafc
ACC = (231, 76, 60, 255)     # #e74c3c
MUT = (148, 163, 184, 255)   # #94a3b8 приглушённый серый

THEMES = ["halo", "havana", "helium", "hover",
          "hunttech-modern", "hunttech-modern-dark", "hunttech-modern-light"]

SIZE = 200
C = SIZE // 2  # центр


def canvas():
    img = Image.new("RGBA", (SIZE, SIZE), BG)
    return img, ImageDraw.Draw(img)


def ring(d, cx, cy, r, fill=FG, width=8):
    d.ellipse([cx - r, cy - r, cx + r, cy + r], outline=fill, width=width)


def save(img, name):
    out = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "..",
                       "modules", "web", "themes", name)
    # файл кладётся в <theme>/icons/dictionaries/<name>.png
    d = os.path.join(out, "icons", "dictionaries")
    os.makedirs(d, exist_ok=True)
    img.save(os.path.join(d, name + ".png"))


# ---- iteraction: две круговые стрелки цикла ----
def iteraction():
    img, d = canvas()
    ring(d, C, C, 62, FG, 10)
    # стрелки по кругу
    for ang in (45, 225):
        a = math.radians(ang)
        x1 = C + 62 * math.cos(a)
        y1 = C + 62 * math.sin(a)
        x2 = C + 62 * math.cos(a + 0.55)
        y2 = C + 62 * math.sin(a + 0.55)
        # треугольник-наконечник
        tip = (x1, y1)
        base1 = (C + 52 * math.cos(a + 0.18), C + 52 * math.sin(a + 0.18))
        base2 = (C + 52 * math.cos(a - 0.18), C + 52 * math.sin(a - 0.18))
        d.polygon([tip, base1, base2], fill=FG)
    d.ellipse([C - 16, C - 16, C + 16, C + 16], fill=ACC)
    return img


# ---- specialisation: мишень ----
def specialisation():
    img, d = canvas()
    ring(d, C, C, 70, FG, 8)
    ring(d, C, C, 46, FG, 8)
    ring(d, C, C, 22, FG, 8)
    d.ellipse([C - 8, C - 8, C + 8, C + 8], fill=ACC)
    return img


# ---- country: глобус с меридианами и пин-маркером ----
def country():
    img, d = canvas()
    ring(d, C, C, 62, FG, 8)
    d.line([C, C - 62, C, C + 62], fill=FG, width=6)      # вертикаль
    d.line([C - 62, C, C + 62, C], fill=FG, width=6)      # горизонталь
    d.arc([C - 62, C - 34, C + 62, C + 34], 0, 180, fill=MUT, width=5)
    d.arc([C - 62, C - 34, C + 62, C + 34], 180, 360, fill=MUT, width=5)
    d.arc([C - 34, C - 62, C + 34, C + 62], 90, 270, fill=MUT, width=5)
    d.arc([C - 34, C - 62, C + 34, C + 62], -90, 90, fill=MUT, width=5)
    # пин сверху
    d.polygon([(C - 14, C - 88), (C + 14, C - 88), (C, C - 58)], fill=ACC)
    d.ellipse([C - 9, C - 97, C + 9, C - 79], fill=ACC)
    return img


# ---- city: силуэты зданий ----
def city():
    img, d = canvas()
    # левое здание
    d.rectangle([C - 84, C - 46, C - 30, C + 62], fill=FG)
    d.rectangle([C - 74, C - 78, C - 40, C - 46], fill=FG)
    # окна левого
    for dx in (-68, -46):
        for dy in (-30, 0, 30):
            d.rectangle([dx, C + dy - 12, dx + 10, C + dy + 12], fill=BG)
    # среднее высокое
    d.rectangle([C - 24, C - 70, C + 26, C + 62], fill=MUT)
    for dx in (-14, 6):
        for dy in (-52, -24, 4, 32):
            d.rectangle([dx, C + dy - 10, dx + 8, C + dy + 10], fill=BG)
    # правое здание
    d.rectangle([C + 34, C - 28, C + 86, C + 62], fill=FG)
    for dx in (44, 66):
        for dy in (-14, 12):
            d.rectangle([dx, C + dy - 10, dx + 10, C + dy + 10], fill=BG)
    return img


# ---- region: карта с дорогами и пин ----
def region():
    img, d = canvas()
    # контур карты-плашки
    d.polygon([(C - 78, C - 62), (C - 12, C - 74), (C + 66, C - 46),
               (C + 80, C + 30), (C + 20, C + 66), (C - 60, C + 52),
               (C - 82, C + 10)], outline=FG, width=8)
    # дороги
    d.line([C - 40, C - 50, C + 30, C + 40], fill=MUT, width=7)
    d.line([C - 60, C + 20, C + 50, C - 20], fill=MUT, width=7)
    # пин-точка
    d.ellipse([C - 10, C - 14, C + 10, C + 6], fill=ACC)
    return img


# ---- ownershup: документ с ключом ----
def ownershup():
    img, d = canvas()
    # документ
    d.rounded_rectangle([C - 58, C - 78, C + 58, C + 70], radius=10,
                        outline=FG, width=8)
    d.line([C - 34, C - 40, C + 34, C - 40], fill=FG, width=7)
    d.line([C - 34, C - 10, C + 34, C - 10], fill=FG, width=7)
    d.line([C - 34, C + 20, C + 10, C + 20], fill=FG, width=7)
    # ключ (акцент)
    d.ellipse([C - 10, C + 30, C + 22, C + 62], outline=ACC, width=8)
    d.line([C + 16, C + 40, C + 44, C + 12], fill=ACC, width=8)
    d.line([C + 44, C + 12, C + 56, C + 12], fill=ACC, width=8)
    d.line([C + 40, C + 16, C + 52, C + 28], fill=ACC, width=8)
    return img


# ---- position: бейдж с человечком ----
def position():
    img, d = canvas()
    # бейдж
    d.rounded_rectangle([C - 66, C - 80, C + 66, C + 70], radius=16,
                        outline=FG, width=8)
    # человечек
    d.ellipse([C - 20, C - 56, C + 20, C - 16], fill=FG)
    d.pieslice([C - 44, C + 8, C + 44, C + 96], 180, 360, fill=FG)
    d.ellipse([C - 16, C + 44, C + 16, C + 60], fill=ACC)
    return img


# ---- file-type: файл с линиями текста ----
def file_type():
    img, d = canvas()
    d.rounded_rectangle([C - 62, C - 78, C + 62, C + 70], radius=10,
                        outline=FG, width=8)
    # загнутый уголок
    d.polygon([(C + 22, C - 78), (C + 62, C - 78), (C + 62, C - 38),
               (C + 22, C - 38)], fill=BG)
    d.line([C + 22, C - 78, C + 62, C - 38], fill=FG, width=8)
    d.line([C + 22, C - 78, C + 22, C - 38], fill=FG, width=8)
    # текст
    for dy in (-16, 8, 32):
        d.line([C - 34, C + dy, C + 30, C + dy], fill=MUT, width=7)
    d.line([C - 34, C + 48, C + 10, C + 48], fill=ACC, width=7)
    return img


# ---- grade: лестница уровней ----
def grade():
    img, d = canvas()
    for i in range(4):
        y = C + 66 - i * 30
        x0 = C - 78 + i * 26
        d.line([x0, y, C + 70, y], fill=FG if i % 2 == 0 else MUT, width=9)
        d.line([x0, y, x0, y - 30], fill=FG if i % 2 == 0 else MUT, width=9)
    d.line([C + 70, C - 24, C + 70, C + 66], fill=FG, width=9)
    d.ellipse([C + 60, C - 40, C + 80, C - 20], fill=ACC)
    return img


# ---- currency: монета с символом валюты ----
def currency():
    img, d = canvas()
    ring(d, C, C, 70, FG, 10)
    # символ "$" из линий
    d.line([C + 6, C - 46, C + 6, C + 46], fill=FG, width=10)
    d.line([C - 14, C - 24, C + 26, C - 24], fill=FG, width=9)
    d.line([C - 14, C + 24, C + 26, C + 24], fill=FG, width=9)
    d.line([C + 6, C - 58, C + 6, C - 40], fill=ACC, width=9)
    d.line([C + 6, C + 40, C + 6, C + 58], fill=ACC, width=9)
    return img


# ---- outstaffing-rates: шкала с процентами ----
def outstaffing_rates():
    img, d = canvas()
    # горизонтальная шкала
    d.rounded_rectangle([C - 76, C - 10, C + 76, C + 10], radius=10,
                        outline=FG, width=8)
    # заполнение
    d.rounded_rectangle([C - 76, C - 10, C + 6, C + 10], radius=10,
                        fill=FG, width=0)
    # стрелка вверх
    d.polygon([(C - 6, C - 52), (C + 26, C - 52), (C + 10, C - 84)],
              outline=FG, width=8)
    d.line([C - 6, C - 52, C + 26, C - 52], fill=FG, width=8)
    d.line([C + 10, C - 84, C + 10, C - 52], fill=FG, width=8)
    d.line([C - 6, C - 52, C + 10, C - 84], fill=FG, width=8)
    d.line([C + 10, C - 84, C + 26, C - 52], fill=FG, width=8)
    # акцентная точка на шкале
    d.ellipse([C + 2, C - 22, C + 22, C - 2], fill=ACC)
    return img


# ---- employee-work-status: часы со статусом ----
def employee_work_status():
    img, d = canvas()
    ring(d, C, C, 70, FG, 10)
    d.line([C, C - 46, C, C], fill=FG, width=9)
    d.line([C, C, C + 34, C + 16], fill=FG, width=9)
    d.ellipse([C - 9, C - 9, C + 9, C + 9], fill=FG)
    # акцентная точка внизу
    d.ellipse([C - 14, C + 46, C + 14, C + 74], fill=ACC)
    return img


# ---- sign-icons: звезда/иконка ----
def sign_icons():
    img, d = canvas()
    cx, cy = C, C - 4
    pts = []
    for i in range(10):
        r = 74 if i % 2 == 0 else 30
        a = math.pi / 2 + i * math.pi / 5
        pts.append((cx + r * math.cos(a), cy + r * math.sin(a)))
    d.polygon(pts, outline=FG, width=8)
    d.ellipse([C - 12, C + 44, C + 12, C + 68], fill=ACC)
    return img


GENERATORS = {
    "iteraction": iteraction,
    "specialisation": specialisation,
    "country": country,
    "city": city,
    "region": region,
    "ownershup": ownershup,
    "position": position,
    "file-type": file_type,
    "grade": grade,
    "currency": currency,
    "outstaffing-rates": outstaffing_rates,
    "employee-work-status": employee_work_status,
    "sign-icons": sign_icons,
}

if __name__ == "__main__":
    for name, fn in GENERATORS.items():
        img = fn()
        for theme in THEMES:
            save(img, theme)
        print("OK", name, img.size, img.mode)
